overlord/ifacestate/helpers.go

```diff
@@ -455,6 +455,150 @@ type connState struct {
 	HotplugKey  snap.HotplugKey `json:"hotplug-key,omitempty"`
 }
 
+type gadgetConnect struct {
+	st   *state.State
+	task *state.Task
+	repo *interfaces.Repository
+
+	instanceName string
+
+	deviceCtx snapstate.DeviceContext
+}
+
+func newGadgetConnect(s *state.State, task *state.Task, repo *interfaces.Repository, instanceName string, deviceCtx snapstate.DeviceContext) *gadgetConnect {
+	return &gadgetConnect{
+		st:           s,
+		task:         task,
+		repo:         repo,
+		instanceName: instanceName,
+		deviceCtx:    deviceCtx,
+	}
+}
+
+// addGadgetConnections adds to newconns any applicable connections
+// from the gadget connections stanza.
+// conflictError is called to handle checkAutoconnectConflicts errors.
+func (gc *gadgetConnect) addGadgetConnections(newconns map[string]*interfaces.ConnRef, conns map[string]*connState, conflictError func(*state.Retry, error) error) error {
+	var seeded bool
+	err := gc.st.Get("seeded", &seeded)
+	if err != nil && err != state.ErrNoState {
+		return err
+	}
+	// we apply gadget connections only during seeding or a remodeling
+	if seeded && !gc.deviceCtx.ForRemodeling() {
```

Comment:
Could this check be moved outside to batchConnectTask to avoid creating of newGadgetConnect  and gadgectConnect.addGadgetConnections alltogether when needed except for seeding/remodeling?

Reply:
newGadgetConnect shouldn't cost a lot, I personally sort of prefer the separation of responsibilities we have by doing the check here

Reply:
Okay, fair enough.